src/uct/tcp/tcp_iface.c

```diff
@@ -131,37 +131,51 @@ static ucs_status_t uct_tcp_iface_event_fd_get(uct_iface_h tl_iface, int *fd_p)
 {
     uct_tcp_iface_t *iface = ucs_derived_of(tl_iface, uct_tcp_iface_t);
 
-    *fd_p = iface->epfd;
-    return UCS_OK;
+    return ucs_event_set_fd_get(iface->event_set,fd_p);
 }
 
-static inline unsigned
-uct_tcp_iface_handle_events(uct_tcp_ep_t *ep, uint32_t epoll_events)
+//static inline unsigned
+//uct_tcp_iface_handle_events(uct_tcp_ep_t *ep, uint32_t event_set_events)
+static void uct_tcp_iface_handle_events(void *callback_data ,
+                                        int event_set_events, void *arg)
 {
-    unsigned count = 0;
+    unsigned count = (unsigned)(uintptr_t)arg;
+    uct_tcp_ep_t *ep = (uct_tcp_ep_t *)callback_data;
 
-    if (epoll_events & EPOLLIN) {
+    if (event_set_events & UCS_EVENT_SET_EVREAD) {
         count += uct_tcp_ep_progress(ep, UCT_TCP_EP_CTX_TYPE_RX);
     }
-    if (epoll_events & EPOLLOUT) {
+    if (event_set_events & UCS_EVENT_SET_EVWRITE) {
         count += uct_tcp_ep_progress(ep, UCT_TCP_EP_CTX_TYPE_TX);
     }
-
-    return count;
 }
 
 unsigned uct_tcp_iface_progress(uct_iface_h tl_iface)
 {
     uct_tcp_iface_t *iface = ucs_derived_of(tl_iface, uct_tcp_iface_t);
-    unsigned read_events   = 0;
-    unsigned count         = 0;
-    struct epoll_event events[UCT_TCP_MAX_EVENTS];
-    int i, nevents, max_events;
-
+    unsigned read_events = 0;
+    unsigned count       = 0;
+    int max_events;
+    unsigned nevents = 0;
+    ucs_status_t status;
+    
     do {
         max_events = ucs_min(iface->config.max_poll - read_events,
                              UCT_TCP_MAX_EVENTS);
 
+        status = ucs_event_set_wait(iface->event_set, max_events, 0,
+                                    uct_tcp_iface_handle_events,
+                                    (void *)(uintptr_t)count, &nevents);
+        if (status != UCS_OK) {
+            if (status == UCS_INPROGRESS) {
+                /* force a new loop iteration */
+                continue;
+            }
+            ucs_error("ucs_event_set_wait(max=%d) failed: %m", max_events);
```

Comment:
we already have error logging inside `ucs_event_set_wait`

Reply:
Removed